docs/sdk/v4/reference/classes/V4Planner.md

```diff
@@ -0,0 +1,145 @@
+[@uniswap/v4-sdk](https://github.com/Uniswap/sdks/tree/main/sdks/v4-sdk) / V4Planner
+
+# Class: V4Planner
+
+Defined in: [utils/v4Planner.ts:167](https://github.com/Uniswap/sdks/blob/c1c9f64f11640c79a680f539823458931629e6ed/sdks/v4-sdk/src/utils/v4Planner.ts#L167)
+
+## Extended by
+
+- [`V4PositionPlanner`](V4PositionPlanner.md)
+
+## Constructors
+
+### new V4Planner()
+
+> **new V4Planner**(): [`V4Planner`](V4Planner.md)
+
+Defined in: [utils/v4Planner.ts:171](https://github.com/Uniswap/sdks/blob/c1c9f64f11640c79a680f539823458931629e6ed/sdks/v4-sdk/src/utils/v4Planner.ts#L171)
+
+#### Returns
+
+[`V4Planner`](V4Planner.md)
+
+## Properties
+
+### actions
+
+> **actions**: `string`
+
+Defined in: [utils/v4Planner.ts:168](https://github.com/Uniswap/sdks/blob/c1c9f64f11640c79a680f539823458931629e6ed/sdks/v4-sdk/src/utils/v4Planner.ts#L168)
+
+---
+
+### params
+
+> **params**: `string`[]
+
+Defined in: [utils/v4Planner.ts:169](https://github.com/Uniswap/sdks/blob/c1c9f64f11640c79a680f539823458931629e6ed/sdks/v4-sdk/src/utils/v4Planner.ts#L169)
+
+## Methods
+
+### addAction()
+
+> **addAction**(`type`, `parameters`): [`V4Planner`](V4Planner.md)
+
+Defined in: [utils/v4Planner.ts:176](https://github.com/Uniswap/sdks/blob/c1c9f64f11640c79a680f539823458931629e6ed/sdks/v4-sdk/src/utils/v4Planner.ts#L176)
+
+#### Parameters
+
+##### type
+
+[`Actions`](../enums/Actions.md)
+
+##### parameters
+
+`any`[]
+
+#### Returns
+
+[`V4Planner`](V4Planner.md)
+
+---
+
+### addSettle()
+
+> **addSettle**(`currency`, `payerIsUser`, `amount`?): [`V4Planner`](V4Planner.md)
+
+Defined in: [utils/v4Planner.ts:213](https://github.com/Uniswap/sdks/blob/c1c9f64f11640c79a680f539823458931629e6ed/sdks/v4-sdk/src/utils/v4Planner.ts#L213)
+
+#### Parameters
+
+##### currency
+
+`Currency`
+
+##### payerIsUser
+
+`boolean`
+
+##### amount
+
+`BigNumber`
+
+#### Returns
+
+[`V4Planner`](V4Planner.md)
+
+---
+
+### addTake()
+
+> **addTake**(`currency`, `recipient`, `amount`?): [`V4Planner`](V4Planner.md)
+
+Defined in: [utils/v4Planner.ts:218](https://github.com/Uniswap/sdks/blob/c1c9f64f11640c79a680f539823458931629e6ed/sdks/v4-sdk/src/utils/v4Planner.ts#L218)
+
+#### Parameters
+
+##### currency
+
+`Currency`
+
+##### recipient
+
+`string`
+
+##### amount
+
+`BigNumber`
+
+#### Returns
+
+[`V4Planner`](V4Planner.md)
+
+---
+
+### addTrade()
+
+> **addTrade**(`trade`, `slippageTolerance`?): [`V4Planner`](V4Planner.md)
+
+Defined in: [utils/v4Planner.ts:183](https://github.com/Uniswap/sdks/blob/c1c9f64f11640c79a680f539823458931629e6ed/sdks/v4-sdk/src/utils/v4Planner.ts#L183)
+
+#### Parameters
+
+##### trade
+
+`Trade`\<`Currency`, `Currency`, `TradeType`\>
```

Comment:
grr this is kind of weird, so on the localhost:3000 its not escaping

the unfortunate thing is, when we upgrade to docusaurus 3 we DO want these characters escaped...

![image](https://github.com/user-attachments/assets/b1a4bfbc-1a0e-4090-8b90-c00da666ead8)

Reply:
Are there any plans to upgrade to 3? 
We could work on a phased overhaul of it.

Reply:
here's the PR for docusaurus3; i would imagine its going in semi soon..

https://github.com/Uniswap/docs/pull/826

---

i wonder if the best plan rn is to strip the escape character in the script (with `sed`), and then we can easily remove it so the script produces docs with escaped characters

Reply:
Will explore it tonight 👍🏻

Reply:
@0xdevant @saucepoint Added step 4 to the script to remove escape characters. To be removed when docusaurus 3 is in.